File: Problems/boj_게임.py
#https://www.acmicpc.net/problem/1072
#Binary Search, Parametric Search

x, y = map(int, input().split())

z = y*100 // x #int로 씌우는 것보다 정확

result = 0
if z>=99:
    print(-1)
else:
    s = 1
    e = int(1e9)
    while s<=e:
        mid = (s+e) // 2
        newz = (y+mid) * 100 // (x+mid)
        if newz != z: # 계속 이기기만 하니깐 newz>z일 것이다.
            result = mid # 정답인 조건에서의 mid를 답으로 지정
            e = mid-1
        else:
            s = mid + 1
    print(result)

File: Problems/cpp/boj_게임.cpp
//https://www.acmicpc.net/problem/1072
//Binary Search

#include <iostream>

using namespace std;

long x, y; //c++는 자료형에 주의! long도 4byte인데 왜 되지? -> 채점 서버가 리눅스인듯?
int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    cin >> x >> y;

    int z = 100*y / x;
    int s=1; int e=1e9;

    int result=0;
    while (s<=e){
        int mid = (s+e)/2;

        int newz = 100*(y+mid) / (x+mid);
        if (newz>z){
            e = mid-1;
            result = mid;
        }
        else
            s = mid+1;
    }
    if (!result) cout << -1;
    else cout << result;
}